_sources/rust-2019-software-practices.md

```diff
@@ -0,0 +1,13 @@
+---
+title: "Rust 2019 (6): Software practices"
+---
+
+# Rust 2019 (6): Software practices
+
+Another thing I highlighted in [Rust 2018: The year we learn Rust 1.0](https://yakshav.es/rust-2018) is the need to develop software practices at scale. I have a positive view on our progress there.
+
+Crucial tooling like `cargo generate` are in a good shape, a great example for this are the project setups in the [embedded book](TODO: Link). It allows us to pre-ship reasonable application structures. I'm a huge fan of it and hope for a wider adoption. In my book, `cargo generate` should replace `cargo new --bin` in most cases in the future.
```

Comment:
[cargo-generate](https://crates.io/crates/cargo-generate) is not inbuilt, so I think you should link to the project.